contrib/generate_precompile.jl

```diff
@@ -329,9 +334,16 @@ function generate_precompile_statements()
                 write(ptm, l, "\n")
                 readuntil(output_copy, "\n")
                 # wait for the next prompt-like to appear
-                # NOTE: this is rather inaccurate because the Pkg REPL mode is a special flower
                 readuntil(output_copy, "\n")
-                readuntil(output_copy, "> ")
+                strbuf = ""
+                while true
+                    strbuf *= String(readavailable(output_copy))
+                    occursin(JULIA_PROMPT, strbuf) && break
+                    occursin(PKG_PROMPT, strbuf) && break
+                    occursin(SHELL_PROMPT, strbuf) && break
+                    occursin(HELP_PROMPT, strbuf) && break
+                    sleep(0.1)
```

Comment:
Why is this `sleep` needed?

Reply:
Oh, I assume that `readavailable` blocks on no data but that might not be the case?

Reply:
Maybe it isn't, I didn't test it without. I just thought spinning without wait if the terminal is stalling output wasn't necessary

Reply:
Maybe we need a `readuntil` function that takes a regex

Reply:
yeah, I looked for that while doing this